pkg/util/patroni/patroni.go

```diff
@@ -0,0 +1,76 @@
+package patroni
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+const (
+	failoverPath = "/failover"
+	apiPort      = 8008
+	timeout      = 30 * time.Second
+)
+
+// Interface describe patroni methods
+type Interface interface {
+	Failover(master *v1.Pod, candidate string) error
+}
+
+// Patroni API client
+type Patroni struct {
+	httpClient *http.Client
+	logger     *logrus.Entry
+}
+
+// New create patroni
+func New(logger *logrus.Entry) *Patroni {
+	cl := http.Client{
+		Timeout: timeout,
+	}
+
+	return &Patroni{
+		logger:     logger,
+		httpClient: &cl,
+	}
+}
+
+func (p *Patroni) apiURL(masterPod *v1.Pod) string {
+	return fmt.Sprintf("http://%s:%d", masterPod.Status.PodIP, apiPort)
+}
+
+// Failover does manual failover via patroni api
+func (p *Patroni) Failover(master *v1.Pod, candidate string) error {
+	buf := &bytes.Buffer{}
+	err := json.NewEncoder(buf).Encode(map[string]string{"leader": master.Name, "member": candidate})
+
+	request, err := http.NewRequest(http.MethodPost, p.apiURL(master)+failoverPath, buf)
```

Comment:
a nitpick: I'd add failoverpath to the apiURL method.

Reply:
what if we want to use other api endpoints?